tailwind.config.js

```diff
@@ -1,16 +1,40 @@
 const colors = require('tailwindcss/colors')
 
+function exposeColorsToCssVars({ addBase, theme }) {
+  function extractColorVars(colorObj, colorGroup = '') {
+    return Object.keys(colorObj).reduce((vars, colorKey) => {
+      const value = colorObj[colorKey];
+
+      const newVars =
+        typeof value === 'string'
+          ? { [`--color${colorGroup}-${colorKey}`]: value }
+          : extractColorVars(value, `-${colorKey}`);
+
+      return { ...vars, ...newVars };
+    }, {});
+  }
+
+  addBase({
+    ':root': extractColorVars(theme('colors')),
+  });
+}
+
 module.exports = {
   mode: 'jit',
   purge: [
     './src/**/*.js',
     './src/**/*.cljs',
     './resources/**/*.html',
   ],
-  plugins: [require('@tailwindcss/ui')],
+  plugins: [
+    require('@tailwindcss/ui'),
+    exposeColorsToCssVars],
   darkMode: 'class',
   theme: {
     extend: {
+      fontSize: {
+        '2xs': ['0.625rem', '0.875rem']
+      },
```

Comment:
I believe this is the [suggested way](https://tailwindcss.com/docs/font-size#using-custom-values) of extending tailwind's font sizes. This would also allow us to use `text-2xs` with the `@apply` directive.